sources/vega/commands/v1/transaction.proto

```diff
@@ -84,4 +84,13 @@ message Transaction {
   // A version of the transaction, to be used in the future in case changes are implemented
   // to the Transaction format.
   uint32 version = 2000;
+
+  // Proof of work contains the random transaction id used by the client and the nonce. 
+  ProofOfWork pow = 3000;
 }
+
+message ProofOfWork {
+  string tx_id = 1;
```

Comment:
Is this transaction ID not a bit redundant? I guess any verifier would not want to trust this, but calculate the `tx_id` themselves of the `input_data`

Reply:
Ah I guess this is actually what the spec calls the `tid`?

Reply:
I got myself confused. The spec says:

>  The user then attaches the PoW to a transaction, and sends it of together with `x` and `H(b)`.

So I guess it should be `nonce` (`x`) and `block_hash`?